term-utils/scriptreplay.c

```diff
@@ -122,10 +125,15 @@ appendchr(char *buf, size_t bufsz, int c)
 }
 
 static int
-setterm(struct termios *backup)
+setterm(struct termios *backup, int *saved_flag)
 {
 	struct termios tattr;
 
+	*saved_flag = fcntl(STDIN_FILENO, F_GETFL);
+	if (*saved_flag == -1)
+		err(EXIT_FAILURE, _("unexpected fcntl failure"));
+	fcntl(STDIN_FILENO, F_SETFL, O_NONBLOCK);
```

Comment:
@jNullj  -- Should this be implemented as:
```
     fcntl(STDIN_FILENO, F_SETFL, saved_flag | O_NONBLOCK);
```
It means adding O_NONBLOCK to exiting flags rather than resetting all.

Reply:
Good catch! The current code does rewrite the flags in that int insted of adding the nonblock flag.
My only addition would be using `*saved_flag` here as `saved_flag` is *int and not int.

```suggestion
	fcntl(STDIN_FILENO, F_SETFL, *saved_flag | O_NONBLOCK);
```

Should i make a new commit to this branch and open a PR to add this fix?

Reply:
Fixed by https://github.com/util-linux/util-linux/commit/41e7686c9ad1ea7892b9d8941c266869bf6a28dd